filelog.go

```diff
@@ -136,28 +177,58 @@ func (w *FileLogWriter) Rotate() {
 	w.rot <- true
 }
 
-// If this is called in a threaded context, it MUST be synchronized
-func (w *FileLogWriter) intRotate() error {
-	// Close any log file that may be open
-	if w.file != nil {
-		fmt.Fprint(w.file, FormatLogRecord(w.trailer, &LogRecord{Created: time.Now()}))
-		w.file.Close()
+// Generate the next filename for rotation using integer suffix
+func (w *FileLogWriter) nextIntegerFilename(filename string) (string, error) {
+	for i := 1; i <= 999; i++ {
```

Comment:
I guess it depends on what happens when we hit 999 files.  If it's "we stop logging" then that's a problem. If it's "we keep logging but we stop rotating" that's not really great but I think we can live with it.